app/controllers/redirect_controller.rb

```diff
@@ -1,23 +1,23 @@
 class RedirectController < ApplicationController
   def item
     return render_404 ActiveRecord::RecordNotFound unless is_uuid
-    file = find_item
+    file = find_item_id
     return render_404 ActiveRecord::RecordNotFound unless file != nil
```

Comment:
unless file != nil should be changed to "if file.nil?" and that seem to fix the issue to redirect items that don't exist.